### PR TITLE
feat(manager): adding popup for choosing static and containerized deployment on application listing page

### PR DESCRIPTION
**Pull request by nikhitamore**
…loyment on application listing page

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+interface CreateStaticAppProps {
+  onClose: () => void;
+  propertyIdentifier: string;
+}
+
+const schema = yup.object({
+  name: yup.string().required().label('Application Name'),
+  path: yup
+    .string()
+    .matches(/^[a-zA-Z0-9/-]+$/, 'Only letters, numbers, forward slash and dashes are allowed')
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:30]: kindly put it in a variable

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+  });
+
+  const webProperties = useGetWebPropertyGroupedByEnv(propertyIdentifier);
+  const webPropertiesKeys = Object.keys(webProperties.data || {});
+  const createStaticSpa = useCreateStaticApp();
+
+  const onSubmit = async (data: FormData) => {
+    try {
+      const formData = new FormData();
+      formData.append('propertyIdentifier', propertyIdentifier);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:75]: `propertyIdentifier` is not needed

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+      formData.append('name', data.name);
+      formData.append('path', data.path);
+      formData.append('env', data.env);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:78]: `env` is not needed

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+      await createStaticSpa.mutateAsync(formData);
+      onClose();
+      toast.success('Deployed application successfully');
+    } catch (error) {
+      if (error instanceof AxiosError && error.response && error.response.status === 403) {
+        toast.error("You don't have access to perform this action");
+        onClose();
+      } else if (error instanceof AxiosError && error.response && error.response.status === 400) {
+        toast.error(
+          "Given Image URL doesn't exist on the source registry, please provide a valid image URL."
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:94]: `Given Image URL doesn't exist on the source registry, please provide a valid image URL.`

Please change it to `Please provide the valid input file`

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+  };
+
+  const handleFileChange = async (value: File | string, fileName: string) => {
+    setValue('upload', [value], { shouldValidate: true });
+    setFileValue(value);
+    setFilename(fileName);
+    await trigger('upload'); // Manually trigger validation on file change
+  };
+
+  // const watchAllFields = watch(); // Watch all fields
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:109]: Please remove this line

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx (hunk 2 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx:76]: This is required at parent level at application listing page

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx (hunk 4 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx]: This is required at parent level at application listing page

**Review by soumyadip007 — APPROVED** [review]
LGTM

File: packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx (hunk 5 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx]: done

**Pull request merged**